### ISSUE TITLE
Invalid redirects: administrator cannot change user's picture details without "page not found" errors

### ISSUE THREAD

**Issue by judgej** [vanilla/vanilla#561]
Many of the redirects in class.profilecontroller.php (vanilla 2.0.9) go to invalid pages. The URLs that do not work generally have the form:

dashboard/profile/{UserID}

or

dashboard/profile/{User Name in URL format}

Neither of these work. The working URL to return to an arbitrary user's account settings is:

profile/{UserID}/{User Name in URL format}

There are about half a dozen places where this needs to be changed. Once changed, editing user's photos is much easier. Examples of changes are lines

79 remove 'dashboard'
129 remove leading '/'
309 remove 'dashboard', add URL-formatted Name
364 remove 'dashboard', add UserID (I think)
374 remove 'dashboard', add URL-formatted Name
473 remove 'dashboard', add UserID
etc.

Basically it needs a thorough looking over.
